numpy/core/src/umath/logical_gufuncs.c.src

```diff
@@ -0,0 +1,304 @@
+        short_ ## NAME,                                 \
+        ushort_ ## NAME,                                \
+        int_ ## NAME,                                   \
+        uint_ ## NAME,                                  \
+        long_ ## NAME,                                  \
+        ulong_ ## NAME,                                 \
+        longlong_ ## NAME,                              \
+        ulonglong_ ## NAME,                             \
+        float_ ## NAME,                                 \
+        double_ ## NAME,                                \
+        longdouble_ ## NAME,                            \
+        cfloat_ ## NAME,                                \
+        cdouble_ ## NAME,                               \
+        clongdouble_ ## NAME,                           \
+    }
+
+
+GUFUNC_FUNC_ARRAY(all_eq);
+GUFUNC_FUNC_ARRAY(all_ne);
+GUFUNC_FUNC_ARRAY(all_lt);
+GUFUNC_FUNC_ARRAY(all_le);
+GUFUNC_FUNC_ARRAY(all_gt);
+GUFUNC_FUNC_ARRAY(all_ge);
+GUFUNC_FUNC_ARRAY(any_eq);
+GUFUNC_FUNC_ARRAY(any_ne);
+GUFUNC_FUNC_ARRAY(any_lt);
+GUFUNC_FUNC_ARRAY(any_le);
+GUFUNC_FUNC_ARRAY(any_gt);
+GUFUNC_FUNC_ARRAY(any_ge);
+
+
+/* -------------------------------------------------------------------------- */
+/* Create type arrays for each gufunc, which are all identical*/
+static char types[48] = {NPY_BYTE, NPY_BYTE, NPY_BOOL,
+                         NPY_UBYTE, NPY_UBYTE, NPY_BOOL,
+                         NPY_SHORT, NPY_SHORT, NPY_BOOL,
+                         NPY_USHORT, NPY_USHORT, NPY_BOOL,
+                         NPY_INT, NPY_INT, NPY_BOOL,
+                         NPY_UINT, NPY_UINT, NPY_BOOL,
+                         NPY_LONG, NPY_LONG, NPY_BOOL,
+                         NPY_ULONG, NPY_ULONG, NPY_BOOL,
+                         NPY_LONGLONG, NPY_LONGLONG, NPY_BOOL,
+                         NPY_ULONGLONG, NPY_ULONGLONG, NPY_BOOL,
+                         NPY_FLOAT, NPY_FLOAT, NPY_BOOL,
+                         NPY_DOUBLE, NPY_DOUBLE, NPY_BOOL,
+                         NPY_LONGDOUBLE, NPY_LONGDOUBLE, NPY_BOOL,
+                         NPY_CFLOAT, NPY_CFLOAT, NPY_BOOL,
+                         NPY_CDOUBLE, NPY_CDOUBLE, NPY_BOOL,
+                         NPY_CLONGDOUBLE, NPY_CLONGDOUBLE, NPY_BOOL};
+
+
+/* -------------------------------------------------------------------------- */
+/* define name and docstring for each gufunc*/
+
+typedef struct gufunc_descriptor_struct {
+    char *name;
+    char *doc;
+    PyUFuncGenericFunction *funcs;
+} GUFUNC_DESCRIPTOR_t;
+
+static GUFUNC_DESCRIPTOR_t gufunc_descriptors [] = {
+    {
+        "all_equal",
+        "Return True if x1 == x2 for all elements along the last axis, False\n"
+        "otherwise.  Similar to (x1==x2).all(axis=-1), except the last dimension\n"
+        "of x1 and x2 must be equal and greater than 1.  This function short\n"
+        "circuits after the first nonequal element."
```

Comment:
Should be clearer, e.g. short circuits on the first nonequal within the reduced axis. I think it might also not be the case when the axis is larger than the iterator buffer size due to limitations of the current implementation
Might be best to not document that behaviour yet.